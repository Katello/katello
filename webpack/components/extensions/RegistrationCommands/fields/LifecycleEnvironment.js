import React from 'react';
import PropTypes from 'prop-types';

import {
  FormGroup,
  FormSelectOption, FormSelect,
} from '@patternfly/react-core';

import { sprintf, translate as __ } from 'foremanReact/common/I18n';
import { noop } from 'foremanReact/common/helpers';

const LifecycleEnvironment = ({
  pluginValues, onChange, isLoading,
  hostGroupEnvironment, lifecycleEnvironments,
}) => (
  <FormGroup
    label={__('Lifecycle environment')}
    fieldId="reg_katello_lce"
    helperText={hostGroupEnvironment && sprintf('From host group: %s', hostGroupEnvironment)}
  >
    <FormSelect
      value={pluginValues?.lifecycleEnvironmentId}
      onChange={v => onChange({ lifecycleEnvironmentId: v })}
      className="without_select2"
      id="reg_katello_lce"
      isDisabled={isLoading || lifecycleEnvironments.length === 0}
    >
      <FormSelectOption
        value=""
        label={lifecycleEnvironments.length === 0 ? __('No Lifecycle environment to select') : ''}
      />
      {lifecycleEnvironments.map(lce => (
        <FormSelectOption key={lce.id} value={lce.id} label={lce.name} />
      ))}
    </FormSelect>
  </FormGroup>
);

LifecycleEnvironment.propTypes = {
  pluginValues: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  onChange: PropTypes.func,
  hostGroupEnvironment: PropTypes.string,
  lifecycleEnvironments: PropTypes.array, // eslint-disable-line react/forbid-prop-types
  isLoading: PropTypes.bool,
};

LifecycleEnvironment.defaultProps = {
  onChange: noop,
  isLoading: false,
  hostGroupEnvironment: '',
  lifecycleEnvironments: [],
  pluginValues: {},

};

export default LifecycleEnvironment;
