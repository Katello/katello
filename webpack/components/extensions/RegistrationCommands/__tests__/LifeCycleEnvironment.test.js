import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';
import LifecycleEnvironment from '../fields/LifecycleEnvironment';

const fixtures = {
  renders: {
    pluginValues: {}, isLoading: false, onChange: () => {}, hostGroupEnvironment: '', lifecycleEnvironments: [],
  },
};

describe('LifecycleEnvironment', () =>
  testComponentSnapshotsWithFixtures(LifecycleEnvironment, fixtures));
