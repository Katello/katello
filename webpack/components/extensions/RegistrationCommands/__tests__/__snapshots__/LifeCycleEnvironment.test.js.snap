// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LifecycleEnvironment renders 1`] = `
<FormGroup
  fieldId="reg_katello_lce"
  helperText=""
  label="Lifecycle environment"
>
  <FormSelect
    className="without_select2"
    id="reg_katello_lce"
    isDisabled={true}
    isIconSprite={false}
    isRequired={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    ouiaSafe={true}
    validated="default"
    value=""
  >
    <FormSelectOption
      label="No Lifecycle environment to select"
      value=""
    />
  </FormSelect>
</FormGroup>
`;
